controllers/front/ProductController.php

```diff
@@ -1026,6 +1027,10 @@ function ($elem) use ($checkedIdProductAttribute) {
             }
         }
 
+        if ($checkedIdProductAttribute === 0) {
```

Comment:
I don't think this should be performed in `tryToGetAvailableIdProductAttribute` Considering the name of the method it does what it's supposed to do, which is checking that the combination is available in stock
Rather the problem here is that `$checkedIdProductAttribute` is equal to zero so this method can't fix anything

I'm looking at where it would make the most sense to get this default id

Reply:
I'd say it should be in `getIdProductAttributeByRequestOrGroup` but then the name wouldn't make sense any more as well
I'd say we should add a new function `getIdProductAttributeByGroupRequestOrDefault` 😅 (the order reflects more the priority then the previous function) and replace all previous calls to `getIdProductAttributeByRequestOrGroup`

Reply:
It still try to get an available attribute id. If there is no one, the getDefaultAttribute will also return 0 :)

Reply:
Variable `$checkedIdProductAttribute` is equal to zero because of the request doesn't have any information about the attribute id. This is why in this context, when you try to get available product attribute id, it should test for the default one when it's possible.

Reply:
Something like that

```php
    /**
     * Return id_product_attribute by id_product_attribute group parameter,
     * or request parameter, or the default attribute as a fallback.
     *
     * @return int|null
     *
     * @throws PrestaShopException
     */
    private function getIdProductAttributeByGroupOrRequestOrDefault()
    {
        $idProductAttribute = $this->getIdProductAttributeByGroup();
        if (null === $idProductAttribute) {
            $idProductAttribute = (int) Tools::getValue('id_product_attribute');
        }
        if (0 === $idProductAttribute) {
            $idProductAttribute = (int) Product::getDefaultAttribute($this->product->id);
        }

        return $this->tryToGetAvailableIdProductAttribute($idProductAttribute);
    }
```

Reply:
Although the naming is lame 😅

Reply:
Yes, I got the problem of the attribute missing in the request, I'm just saying the fallback shouldn't be managed in `tryToGetAvailableIdProductAttribute`

Reply:
Oh, I now understand why I hate this part of code :sweat_smile: Thanks for explanation :+1: